app/models/manageiq/providers/vmware/infra_manager/provision/configuration.rb

```diff
@@ -5,11 +5,28 @@ module ManageIQ::Providers::Vmware::InfraManager::Provision::Configuration
   include_concern 'Network'
   include_concern 'Disk'
 
-  def reconfigure_hardware_on_destination?
+  def reconfigure_container_on_destination?
     # Do we need to perform a post-clone hardware reconfigure on the new VM?
     [:cpu_limit, :memory_limit, :cpu_reserve, :memory_reserve].any? do |k|
       return false unless options.key?(k)
       destination.send(k) != options[k]
     end
   end
+
+  def reconfigure_disk_on_destination?(destination)
+    default_size = destination.hardware.disks.find_by(:device_type => "disk").size / (1024**3)
```

Comment:
We're going to need to decide how we want to handle multiple disks.

I could see by default this only works with the boot disk, but we could allow passing additional disk sizes by specifying the device_name/location

Reply:
> I gave this a quick test on a live VC and confirmed that this will reconfigure the resulting VM to resize the disk prior to powering the VM on.
> 
> @OrGur1987 I'm thinking we'll want to add the `:disk_size` option to the other `content/miq_dialogs` and in the future we can have it as an option on the normal provision dialog as well.

I added it to the rest of the yamls, but not sure how to check that they work.
do these yamls have their own hamls that must be adjusted, or do the changes I already made in ui-classic cover this?

> We're going to need to decide how we want to handle multiple disks.
> 
> I could see by default this only works with the boot disk, but we could allow passing additional disk sizes by specifying the device_name/location

do you want me to add this in this PR or is it for a future enhancement?

Reply:
I think we need to deal with it in this PR, since the return order of records isn't guaranteed using find_by with multiple matching records can produce erratic results and we could end up making a second or third disk dramatically smaller by accident.

I think the quickest fix for this in the short term is to only allow changing the disk size if there is 1 disk on the template.  We can doc this and work on improving it going forward.

The intermediate fix would be to sort disks by `Hard Disk X` and apply the disk_size to the "first" disk only.

Longer term allowing users to specify different disk sizes for each of the disks on the template would be nice.

Reply:
thanks!
I'll work on the short term fix, can't promise I'll be available to contribute to the intermediate and long term ones :)

Reply:
Yeah of course not a problem this is a huge start we can pick it up from there

Reply:
`reconfigure_disk_on_destination?` now checks if there's more than one hard disk, and if so logs that it's not currently supported and returns `false`

Reply:
hi @agrare 
is the recent update ok?